### PR TITLE
Pull in upstream updates

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request pulls in upstream updates from [cisagov/skeleton-python-library] which effectively re-skeletonizes the project at this point.

## 💭 Motivation and context ##

This project does not currently receive automatic updates from our [Lineage](https://github.com/cisagov/action-lineage) process and no longer follows our current Python library development style as laid down in [cisagov/skeleton-python-library]. Being up to date and staying that way semi-automatically is always the best way for our actively used repositories to be.

## 🧪 Testing ##

Automates tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Add a tag or create a release.

[cisagov/skeleton-python-library]: https://github.com/cisagov/skeleton-python-library

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]
Strong work!

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        return
+    skip_slow = pytest.mark.skip(reason="need --runslow option to run")
+    for item in items:
+        if "slow" in item.keywords:
+            item.add_marker(skip_slow)
+
+
+@pytest.fixture(autouse=True)
+def test_fs(request, fs):
+    """Set up the contents of the fake fillesystem."""
```
Comment (dav3r) [inline tests/conftest.py]: Typo:
```suggestion
    """Set up the contents of the fake filesystem."""
```

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    fs.add_real_file(
+        os.path.join(request.fspath.dirname, TEST_CONFIGURATION),
+        target_path="good_configuration.yml",
+    )
+    with open("good_configuration.yml", "r") as good_config:
+        base_config = yaml.load(good_config, Loader=yaml.SafeLoader)
+
```
Comment (dav3r) [inline tests/conftest.py:51]: I'd like to see a brief comment above each of these file-creation sections explaining the purpose of each test file being created, e.g. `# Create a test config file with no "database" section`

File: tests/test_mongo_db_from_config.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            "no_database_configuration.yml"
+        )
+        assert (
+            db_connection.list_collection_names() == []
+        ), "Testing database should have no collections"
+    assert e_info.value.args[0] == "database", "Expected a missing 'database' key"
+
+
+def test_db_from_config_bad_yaml_no_db_uri():
+    """Test that YAML parsing fails due to a missing database uri entry."""
```
Comment (dav3r) [inline tests/test_mongo_db_from_config.py]: I understand why you put "uri" in lower case here, but I think proper capitalization is justified if you are not quoting "uri" (as you do in the assertion below).  Feel free to ignore this if you feel strongly about it:
```suggestion
    """Test that YAML parsing fails due to a missing database URI entry."""
```

**Review by dav3r — APPROVED** [review]
Nice work here!  🐲 

I noted a few small items for you consideration.

File: tests/test_mongo_db_from_config.py (hunk 3 above)
Comment (mcdonnnj) [reply tests/test_mongo_db_from_config.py]: Please see [4061322](https://github.com/cisagov/mongo-db-from-config/pull/8/commits/4061322fd641ae5974aaf75496c28a32c018081c).

File: tests/conftest.py (hunk 1 above)
Comment (mcdonnnj) [reply tests/conftest.py]: Please see [4061322](https://github.com/cisagov/mongo-db-from-config/pull/8/commits/4061322fd641ae5974aaf75496c28a32c018081c).

File: tests/conftest.py (hunk 2 above)
Comment (mcdonnnj) [reply tests/conftest.py:51]: Please see [d6a1955](https://github.com/cisagov/mongo-db-from-config/pull/8/commits/d6a19554771ff5f964061871b1b04a1cf79fb525) and let me know what you think.

Comment (dav3r) [reply tests/conftest.py:51]: That is perfect, thanks! 👍

**Pull request merged**